### PR TITLE
Enable RP registration

### PR DESCRIPTION
**Pull request by amarzavery**

### CODE REVIEW

File: src/CLU/Commands.ResourceManager.Common/Generated/ResourceManagementClientExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Rest;
+    using Microsoft.Rest.Azure;
+    using Models;
+
+    public static partial class ResourceManagementClientExtensions
```
Comment (stankovski) [inline src/CLU/Commands.ResourceManager.Common/Generated/ResourceManagementClientExtensions.cs]: Remove unused code

File: src/CLU/Commands.ResourceManager.Common/RPRegistrationDelegatingHandler.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        public ResourceManagementClient ResourceManagementClient { get; set; }
+
+        public RPRegistrationDelegatingHandler(Func<ResourceManagementClient> createClient, Action<string> writeDebug)
+        {
+            registeredProviders = new HashSet<string>();
+            this.writeDebug = writeDebug;
+            this.createClient = createClient;
+        }
+
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
```
Comment (markcowl) [inline src/CLU/Commands.ResourceManager.Common/RPRegistrationDelegatingHandler.cs]: line breaks

File: src/CLU/Commands.ResourceManager.Common/RPRegistrationDelegatingHandler.cs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                            TestMockSupport.Delay(1000);
+                        } while (provider.RegistrationState != RegistrationState.Registered.ToString());
+                        writeDebug(string.Format(Resources.ResourceProviderRegisterSuccessful, providerName));
+                    }
+                    catch (Exception e)
+                    {
+                        writeDebug(string.Format(Resources.ResourceProviderRegisterFailure, providerName, e.Message));
+                        // Ignore RP registration errors.
+                    }
+
```
Comment (markcowl) [inline src/CLU/Commands.ResourceManager.Common/RPRegistrationDelegatingHandler.cs]: explicitly dispose of the previous response here

**Comment by markcowl** [comment]
@amarzavery Unfortunately, closing

**Pull request closed without merging**